Source/Scene/Camera.js

```diff
@@ -801,6 +829,96 @@ define([
+     * The position can be given as either a cartesian or a cartographic. If both are given,
+     * then the cartesian will be used. If neither is given, then the current camera position
+     * will be used.
+     *
+     * @param {Cartesian3} [options.position] The cartesian position of the camera.
+     * @param {Cartographic} [options.positionCartographic] The cartographic position of the camera.
+     * @param {Number} [options.heading] The heading in radians or the current heading will be used if undefined.
+     * @param {Number} [options.pitch] The pitch in radians or the current pitch will be used if undefined.
+     * @param {Number} [options.roll] The roll in radians or the current roll will be used if undefined.
+     */
+    Camera.prototype.setView = function(options) {
+        if (this._mode === SceneMode.MORPHING) {
+            return;
+        }
+
+        options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+
+        var scene2D = this._mode === SceneMode.SCENE2D;
+        var scene3D = this._mode === SceneMode.SCENE3D;
+
+        var heading = defaultValue(options.heading, this.heading);
+        var pitch = scene2D ? CesiumMath.PI_OVER_TWO : defaultValue(options.pitch, this.pitch);
+        var roll = scene2D ? 0.0 : defaultValue(options.roll, this.roll);
+
+        if (scene3D) {
+            roll = -roll;
+        }
+
+        var cartesian = options.position;
+        var cartographic = options.positionCartographic;
+
+        var projection = this._projection;
+        var ellipsoid = projection.ellipsoid;
+
+        if (!defined(cartesian)) {
+            if (defined(cartographic)) {
+                cartesian = ellipsoid.cartographicToCartesian(cartographic, scratchSetViewCartesian);
+            } else {
+                cartesian = Cartesian3.clone(this.positionWC, scratchSetViewCartesian);
+            }
+        }
+
+        var currentTransform = Matrix4.clone(this.transform, scratchSetViewTransform1);
+        var localTransform = Transforms.eastNorthUpToFixedFrame(cartesian, ellipsoid, scratchSetViewTransform2);
+        this.setTransform(localTransform);
+
+        if (scene2D) {
+            Cartesian2.clone(Cartesian3.ZERO, this.position);
+
+            var cartographic2D = ellipsoid.cartesianToCartographic(cartesian, scratchSetViewCartographic);
+            var newLeft = -cartographic2D.height * 0.5;
+            var newRight = -newLeft;
+
+            var frustum = this.frustum;
+            if (newRight > newLeft) {
+                var ratio = frustum.top / frustum.right;
+                frustum.right = newRight;
+                frustum.left = newLeft;
+                frustum.top = frustum.right * ratio;
+                frustum.bottom = -frustum.top;
+            }
+        } else {
+            Cartesian3.clone(Cartesian3.ZERO, this.position);
+        }
+
+        var headingQuaternion = Quaternion.fromAxisAngle(Cartesian3.UNIT_Z, heading + CesiumMath.PI_OVER_TWO, new Quaternion());
```

Comment:
We use scratches above, but not here.  Why?

I'm not sure that this will be a hot spot, but I do expect users to call this every frame to animate the camera.